pkg/querier/cardinality_analysis_handler.go

```diff
@@ -106,7 +107,16 @@ func ActiveSeriesCardinalityHandler(distributor Distributor, limits *validation.
 			return
 		}
 
-		util.WriteJSONResponse(w, activeSeriesResponse{res})
+		var json = jsoniter.ConfigCompatibleWithStandardLibrary
+		bytes, err := json.Marshal(v1.Response{Data: res})
```

Comment:
Should this have a `Status: "success"` field set as well?

Reply:
Prometheus does set that field, however it's not quite clear to me why. Isn't the HTTP status code a good enough indication that the payload is going to be ok/parseable as expected?

Reply:
Sure, I don't think we need the field either, but I think right now it is returning `Status: ""`, which is not useful. So I'd say we either keep it and conform to the prometheus response (which includes setting the error fields), or just return `Data` and nothing else.

Reply:
Ok, makes sense, I'll stick with the custom struct then.